openpype/hosts/maya/plugins/template_loaders/assign_look_placeholder.py

```diff
@@ -0,0 +1,127 @@
+from functools import partial
+
+from maya import cmds
+
+from openpype.lib import (
+    UISeparatorDef,
+    UILabelDef,
+    TextDef,
+    BoolDef
+)
+from openpype.hosts.maya.api.workfile_template_builder import (
+    MayaPlaceholderPlugin,
+)
+from openpype.hosts.maya.api.lib import (
+    get_all_children,
+    assign_look,
+)
+
+
+class AssignLookPlaceholderPlugin(MayaPlaceholderPlugin):
+    """Assign a look subset to members of the placeholder set.
+
+    Creates an objectSet. Any members will get the look assigned with the given
+    subset name if it exists.
+
+    Any containers loaded from other template placeholders will get the look
+    assigned to their loaded containers.
+
+    """
+
+    identifier = "maya.assignlook"
+    label = "Assign Look"
+
+    def get_placeholder_options(self, options=None):
+        options = options or {}
+        return [
+            UISeparatorDef(),
+            UILabelDef(label="<b>Description</b>"),
+            UISeparatorDef(),
+            UILabelDef(
+                label=(
+                    "Creates an objectSet. Any members will get the look\n"
+                    "assigned with the given subset name if it exists.\n\n"
+                    "Any containers loaded from other template placeholders\n"
+                    "will get the look assigned to their loaded containers."
+                    ""
+
+                )
+            ),
+            UISeparatorDef(),
+            UILabelDef(label="<b>Settings</b>"),
+            UISeparatorDef(),
+            TextDef(
+                "subset",
+                label="Subset",
+                tooltip="Look subset to assign to containers loaded by "
+                        "contained placeholders",
+                multiline=False,
+                default=options.get("subset", "lookMain")
+            ),
+            BoolDef(
+                "recurse",
+                label="Recursive",
+                tooltip="Assign look also to potential sub containers / "
+                        "placeholders loaded from the load placeholder.",
+                default=options.get("recurse", False)
+            ),
+        ]
+
+    def create_placeholder(self, placeholder_data):
+        placeholder_data["plugin_identifier"] = self.identifier
+
+        # Create maya objectSet on selection
+        selection = cmds.ls(selection=True, long=True)
+        subset = placeholder_data["subset"]
+        name = "AssignLook_{}".format(subset)
+        node = cmds.sets(selection, name=name)
+
+        self.imprint(node, placeholder_data)
+
+    def populate_placeholder(self, placeholder):
+        callback = partial(self.assign_look, placeholder)
```

Comment:
You don't need `partial`. The event contains placeholder id which should be possible to use to get the placeholder item in the callback from event, or?

Reply:
The on depth processed and finished callbacks are global events of the template builder, they are not per placeholder and thus do not have the placeholder id or a reference to it.

It's also not possible to register a callback with custom user data like e.g. Maya allows (there called `clientData`).

Hence partial is needed to know what placeholder the registered callback should apply to.

Reply:
So the placeholder which triggered the event is not the same placeholder as in this method?